### PR TITLE
feat: support custom api urls

### PR DESCRIPTION
**Pull request by muety**

### CODE REVIEW

**Comment by muety** [comment]
Not sure why the tests are failing, can you give me a hint, @jaebradley ?

**Comment by jaebradley** [comment]
Hi @muety !

Thanks for the contribution - I'm planning on reviewing it this weekend.

As for the integration tests, were you able to get the integration tests to pass locally following the instructions here: https://github.com/jaebradley/wakatime-client#local-development

File: src/index.js
Comment (jaebradley) [inline src/index.js:21]: Do you have a preference for this approach vs. using [default parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters)?

One benefit of using default parameters is that falsy values other than `undefined` would result in some type of error to the client user vs. using the default url.

In terms of browser-compatibility, while default parameters are not supported by IE, [the current babel configuration](https://github.com/jaebradley/wakatime-client/blob/f50c0f8e5f9cc9b7a988c539810fd4eae0282076/.babelrc#L18) should transpile everything down to ES5.

File: README.md
Comment (jaebradley) [inline README.md:68]: Thanks for updating the documentation, do you mind adding a fake code example as well like

```javascript
const client = new WakaTimeClient('some api key', 'https://hakatime-demo.mtx-dev.xyz/api/v1/');
```

**Review by jaebradley — CHANGES_REQUESTED** [review]
What Node version did you use to build this project locally?

[The `.nvmrc` file](https://github.com/jaebradley/wakatime-client/blob/master/.nvmrc) specifies `12.14.1`. If you didn't install this project locally using this Node version, it might explain the large `package-lock.json` diff.

File: src/index.js
Comment (muety) [reply src/index.js:21]: Frankly, I didn't know about default parameters, until now I thought they were only a thing in TypeScript. Thanks for pointing that out! I'll change it.

File: README.md
Comment (muety) [reply README.md:68]: Sure! However, as I'm the maintainer of Wakapi, I'm probably going to put a Wakapi URL instead 😉

**Comment by muety** [comment]
Thanks for your thorough review! In addition, I'd like to have your opinion on the following. 

Neither Hakatime nor Wakapi entirely mimic WakaTime's API. For instance, there is no leaderboard and no goals. Would you suggest to explicitly account for that in some way? For instance, a call to `/users/current/goals` would yield a `404` error as the endpoint doesn't exist in Wakapi and, accordingly, `getMyGoals()` would return a rejected promise. Would that be fine?

**Comment by jaebradley** [comment]
@muety in response to

>  For instance, there is no leaderboard and no goals. Would you suggest to explicitly account for that in some way?

It seems like the Hakatime and Wakapi interfaces do not entirely mirror the Wakatime interface. I think a `404` is reasonable given the case, but even more reasonable to an API users would be to not include the method at all.

**Comment by muety** [comment]
> but even more reasonable to an API users would be to not include the method at all.

I agree, but how would you want to realize that?

**Comment by muety** [comment]
So how do we proceed here?

**Comment by jaebradley** [comment]
@muety thanks for following up - I think the `404`s / rejected `Promise`s are ok for now. My recommendation would be to create separate clients for Hakatime / Wakapi.

I'm happy to accept this PR as-is.

**Comment by muety** [comment]
Alright, so then let's get this merged for now :)

**Comment by jaebradley** [comment]
Thanks for the PR @muety! (hope you don't mind the tweaks I added before merge)

**Pull request merged**
